src/qureg_init.cpp

```diff
@@ -35,8 +35,9 @@ QubitRegister<Type>::QubitRegister()
 
   fusion = false;
 
-  Resize(1UL);
-  state_storage[0] = {1., 0.};
+  Resize(2UL);
+  state[0] = {1., 0.};
+  state[1] = {0., 0.};
 
   if (nprocs > 1) {
```

Comment:
The assert will trigger if the argument-less constructor is called with more than 1 MPI ranks.

The current unit tests are not capturing this failure since they are located in `one_qubit_register_test.hpp` and all its tests are skipped when there is more than 1 MPI rank (see lines 26-27 there).
If you add an identical test "IntializeWithDefault" to `state_initialization_test.hpp` you will see it fails when 2 or more MPI ranks are used.